src/dex/infusion/infusion.ts

```diff
@@ -0,0 +1,695 @@
+            : await this.getSellPricePath(unitVolumeWithFee, [pairParam]);
+
+        const prices =
+          // @ts-expect-error Buy side is not implemented yet
+          side === SwapSide.BUY
+            ? await Promise.all(
+                amountsWithFee.map(amount =>
+                  amount === 0n
+                    ? 0n
+                    : this.getBuyPricePath(amount, [pairParam]),
+                ),
+              )
+            : await Promise.all(
+                amountsWithFee.map(amount =>
+                  amount === 0n
+                    ? 0n
+                    : this.getSellPricePath(amount, [pairParam]),
+                ),
+              );
+
+        const [unitOutWithFee, ...outputsWithFee] = applyTransferFee(
+          [unit, ...prices],
+          side,
+          // This part is confusing, because we treat differently SELL and BUY fees
+          // If Buy, we should apply transfer fee on srcToken on top of dexFee applied earlier
+          // But for Sell we should apply only one dexFee
+          isSell ? transferFees.destDexFee : transferFees.srcFee,
+          isSell ? this.DEST_TOKEN_DEX_TRANSFERS : SRC_TOKEN_PARASWAP_TRANSFERS,
+        );
+
+        return {
+          prices: outputsWithFee,
+          unit: unitOutWithFee,
+          data: {
+            router: this.router,
+            path: [from.address.toLowerCase(), to.address.toLowerCase()],
+            factory: this.factoryAddress,
+            initCode: this.initCode,
+            feeFactor: this.feeFactor,
+            isFeeTokenInRoute: Object.values(transferFees).some(f => f !== 0),
+            pools: [
+              {
+                address: pairParam.exchange,
+                fee: parseInt(pairParam.fee),
+                direction: pairParam.direction,
+              },
+            ],
+          },
+          exchange: this.dexKey,
+          poolIdentifier,
+          gasCost: this.poolGasCost,
+          poolAddresses: [pairParam.exchange],
+        };
+      });
+
+      const resultPools = (await Promise.all(
+        resultPromises,
+      )) as ExchangePrices<UniswapV2Data>;
+      const resultPoolsFiltered = resultPools.filter(item => !!item); // filter null elements
+      return resultPoolsFiltered.length > 0 ? resultPoolsFiltered : null;
+    } catch (e) {
+      if (blockNumber === 0)
+        this.logger.error(
+          `Error_getPricesVolume: Aurelius block manager not yet instantiated`,
+        );
+      this.logger.error(`Error_getPrices:`, e);
+      return null;
+    }
+  }
+
+  async getTopPoolsForToken(
+    tokenAddress: Address,
+    count: number,
+  ): Promise<PoolLiquidity[]> {
+    if (!this.subgraphURL) return [];
+
+    let stableFieldKey = 'isStable';
+
+    // if (this.dexKey.toLowerCase() === 'infusion') {
+    //   stableFieldKey = 'stable';
+    // } else if (this.dexKey.toLowerCase() !== 'infusionv2') {
+    //   stableFieldKey = 'isStable';
+    // }
```

Comment:
@JackieJoo Do we need to keep these commented lines?

Reply:
fixed